{
  try {
    reset(r);
  }
 catch (  IOException e) {
    throw new RuntimeException(e);
  }
}
